### PR TITLE
Edge Notebooks now create a tab-separated file for ML feedback

### PR DESCRIPTION
**Pull request by LedaLima**
Since ML now expects a tab separated file as feedback, this PR creates the necessary files on the UI for each pipeline and copies them back to ML node.

### CODE REVIEW

**Comment by NathanSegerlind** [comment]
crazy question.... is there any reason that so much of the code has to be in an ipython notebook instead of in a plain old .py file?

**Comment by NathanSegerlind** [comment]
also... where the file / line numbers for the change of output delimiter to tabs?

**Comment by daortizh** [comment]
LGTM :+1:

File: oa/flow/ipynb_templates/Edge_Investigation_master.ipynb [hunk 1]
```diff
@@ -203,9 +202,26 @@
...
+    "                            scored_threats.append(row)    \n",
     "                            rowct += 1\n",
     "                        \n",
-    "                    riter.writerow(row) \n",
+    "                    riter.writerow(row)  \n",
+    "                    \n",
+    "        #works on the feedback tab-separated file\n",
+    "        if not os.path.exists(score_fbk):  \n",
+    "            with open(score_fbk, 'w') as feedback:\n",
+    "                wr = csv.writer(feedback, delimiter='\\t', quoting=csv.QUOTE_NONE)   \n",
```
Comment (rabarona) [inline oa/flow/ipynb_templates/Edge_Investigation_master.ipynb:213]: @NathanSegerlind here is one.

File: oa/dns/ipynb_templates/Edge_Investigation_master.ipynb [hunk 2]
```diff
@@ -167,32 +161,37 @@
...
+    "                    if row['dns_qry_name'] == value[1]:  \n",
+    "                        row['dns_sev'] = value[3]                        \n",
+    "                        scored_threats.append(row)  \n",
     "                        rowct += 1\n",
     "                        break\n",
     "                wr.writerow(row)     \n",
+    "                    \n",
+    "        if not os.path.exists(score_fbk):  \n",
+    "            with open(score_fbk, 'w') as feedback:\n",
+    "                wr = csv.DictWriter(feedback, delimiter='\\t', quoting=csv.QUOTE_NONE, fieldnames=reader.fieldnames)            \n",
```
Comment (rabarona) [inline oa/dns/ipynb_templates/Edge_Investigation_master.ipynb:187]: @NathanSegerlind one more.

File: oa/proxy/ipynb_templates/Edge_Investigation_master.ipynb [hunk 3]
```diff
@@ -139,26 +136,34 @@
...
     "                try:\n",
     "                    wr.writerow(row)\n",
     "                except:\n",
     "                    print str(row)\n",
-    "\n",
+    "                    \n",
+    "        #works on the feedback tab-separated file\n",
+    "        if not os.path.exists(score_fbk):  \n",
+    "            with open(score_fbk, 'w') as feedback:\n",
+    "                wr = csv.DictWriter(feedback, delimiter='\\t', quoting=csv.QUOTE_NONE, fieldnames=reader.fieldnames)            \n",
```
Comment (rabarona) [inline oa/proxy/ipynb_templates/Edge_Investigation_master.ipynb:155]: @NathanSegerlind  and this is for Proxy

File: oa/dns/ipynb_templates/Edge_Investigation_master.ipynb (hunk 2 above)
Comment (NathanSegerlind) [reply oa/dns/ipynb_templates/Edge_Investigation_master.ipynb:187]: ok, thanks

**Comment by NathanSegerlind** [comment]
I will create analagous changes in the ml pipeline... to my limited knowledge of the OA layer, this looks good to me

**Comment by rabarona** [comment]
@LedaLima -> looks great DNS Edge_Investigation_master.ipynb with DictReader/Writer, much more cleaner. 
Hopefully Flow gets to the same point some day.

@NathanSegerlind  -> are you talking about DNS, Flow and Proxy or just Proxy? 👯 

LGTM

**Comment by NathanSegerlind** [comment]
proxy already uses tab delimited feedback, new PR updates dns and flow to use tab delimited feedback

https://github.com/Open-Network-Insight/oni-ml/pull/62

**Comment by LedaLima** [comment]
@NathanSegerlind There is not any particular reason to have the code there, eventually we might get to that point. Thanks for the suggestion

**Pull request merged**